ansible/library/generate_golden_config_db.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+# This ansible module is for generate golden_config_db.json
+# Currently, only enable dhcp_server feature and generated related configuration in MX device
+# which has dhcp_server feature.
+
+
+import copy
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+
+DOCUMENTATION = '''
+module: generate_golden_config.py
+author: Yaqiang Zhu (yaqiangzhu@microsoft.com)
+short_description:   Generate golden_config_db.json
+Description:
+        When load_minigraph, SONiC support to use parameter --override_config to add configuration via
+        golden_config_db.json. This module is to generate required /etc/sonic/golden_config_db.json
+    Input:
+        topo_name: Name of current topo
+'''
+
+GOLDEN_CONFIG_DB_PATH = "/etc/sonic/golden_config_db.json"
+TEMP_GOLDEN_CONFIG_DB_PATH = "/tmp/golden_config_db.json"
+TEMP_DHCP_SERVER_CONFIG_PATH = "/tmp/dhcp_server.json"
+
+
+class GenerateGoldenConfigDBModule(object):
+    def __init__(self):
+        self.module = AnsibleModule(argument_spec=dict(topo_name=dict(required=True, type='str')),
+                                    supports_check_mode=True)
+        self.topo_name = self.module.params['topo_name']
+
+    def generate_mx_golden_config_db(self):
+        """
+        If FEATURE table in init_cfg.json contains dhcp_server, enable it.
+        """
+        rc, out, err = self.module.run_command("cat /etc/sonic/init_cfg.json")
+        if rc != 0:
+            self.module.fail_json(msg="Failed to get init_cfg.json: {}".format(err))
+        init_config_obj = json.loads(out)
+        gold_config_db = {}
+        if "FEATURE" not in init_config_obj:
+            return "{}"
+        for feature in init_config_obj["FEATURE"].keys():
+            if feature == "dhcp_server":
+                init_config_obj["FEATURE"]["dhcp_server"]["state"] = "enabled"
+                gold_config_db = {"FEATURE": copy.deepcopy(init_config_obj["FEATURE"])}
+        # To avoid single quotation exists in str which would cut cmd executed by run_command("sudo sh -c 'xxxxxx'")
+        ret = json.dumps(gold_config_db).replace("\'", "dummy_single_quota")
+        return ret.replace("\"", "\\\"").replace("(", "\\(").replace(")", "\\)")
+
+    def generate(self):
+        if self.topo_name == "mx":
+            config = self.generate_mx_golden_config_db()
+        else:
+            config = "{}"
+
+        cmd = ("sudo sh -c 'echo {} > {}'"
```

Comment:
Will it remove any bad config on the DUT if the process fail?
If not, we need another file holder to save this `config`. 
Because if there is an existing `golden_config_db.json`, it will be polluted when fail at line 71.

Reply:
Updated, after write config to file successfully, it wouldn't exist with failure.